components/EditMode/index.js

```diff
@@ -0,0 +1,163 @@
+          ? { ...menuType, checked: !menuType.checked }
+          : menuType
+      )
+    );
+  }
+  const menuCheck = menuTypes
+    .filter((menuType) => menuType.checked)
+    .map((menuTypeChecked) => menuTypeChecked.type);
+
+  async function handleEditLocation(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const locationData = Object.fromEntries(formData);
+
+    const editLocation = {
+      name: locationData.name,
+      location: locationData.location,
+      zeit: locationData.zeit,
+      art: menuCheck,
+      verleihOpt: checked,
+      verleih: locationData.verleih,
+    };
+
+    const response = await fetch(`/api/locations/${data?._id}`, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(editLocation),
+    });
+    if (response.ok) {
+      mutate();
+      handleOnEditMode();
+    }
+  }
+
+  return (
+    <>
+      <form onSubmit={handleEditLocation}>
+        <label htmlFor="title"> Location:</label>
+        <br />
+        <textarea
+          type="text"
+          id="name"
+          name="name"
+          required
+          minlengh="3"
+          maxlengh="50"
+          defaultValue={data?.name}
+          pattern="[0-9A-Za-zА-Яа-яЁё?\s]+"
+        />
+        <br />
+        <label htmlFor="addresse"> Addresse: </label>
+        <br />
+        <textarea
+          type="text"
+          id="location"
+          name="location"
+          required
+          minlengh="3"
+          max="40"
+          maxlengh="200"
+          defaultValue={data?.location}
+          pattern="[0-9A-Za-zА-Яа-яЁё?\s]+"
+        />
+        <br />
+        <label htmlFor="title"> Öffnungszeiten:</label>
+        <br />
+        <textarea
+          type="text"
+          id="zeit"
+          name="zeit"
+          required
+          minlengh="3"
+          maxlengh="50"
+          defaultValue={data?.zeit}
+          pattern="[0-9A-Za-zА-Яа-яЁё?\s]+"
+        />
+        <br />
+        <legend htmlFor="addresse"> MenuArt: </legend>
+        {menuTypes.map((type) => (
+          <label for={type.type} key={type.id}>
+            {type.type}
+            <input
+              type="checkbox"
+              name="type"
+              value={type.type}
+              id={type.id}
+              onChange={() => handleFilter(type.id)}
+              checked={type.checked}
+            />
+          </label>
+        ))}
+        <br />
+        <legend>Verleih Möglichkeit</legend>
+        <input
+          type="radio"
+          name="verleihOpt"
+          value={data?.verleihOpt === true}
+          id="true"
+          checked={checked}
+          onChange={onOptionChange}
+        />
+        <label htmlFor="true">ja</label>
+        <br />
+        <input
+          type="radio"
+          name="verleihOpt"
+          value={data?.verleihOpt === false}
+          id="false"
+          checked={checked}
```

Comment:
I'm not sure about the `checked` attribute here.

Is it correct to pass the same `checked` value to both input fields (_ja_ & _nein_)?
Don't you need to invert the `checked` value for the _nein_ field?
Do you need the `checked` attribute at all?

Like I said, I'm not sure here. You'll have to see for yourself and make sure it's handled correctly.